### PR TITLE
Add GitHub Workflow Replacement for Jenkins job_PostCommit_Java_Examples_Dataflow_V2

### PR DESCRIPTION
**Pull request by aleksandr-dudko**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Examples_Dataflow_V2/
https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Examples_Dataflow_V2_java11/
https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Examples_Dataflow_V2_java17/

How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

by adding a comment to the PR

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      matrix:
+        job_name: [beam_PostCommit_Java_Examples_Dataflow_V2_Java]
+        job_phrase_1: [Run Java ]
+        job_phrase_2: [Examples on Dataflow Runner V2]
+        java_version: ['11', '17']
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      (startswith(github.event.comment.body, 'Run Java') &&
+         endswith(github.event.comment.body, 'Examples on Dataflow Runner V2'))
```
Comment (damccorm) [inline .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml:69]: I think we should use `contains` so we are tolerant of trailing whitespace/other chars

**Review by damccorm — COMMENTED** [review]
Mostly looks good, had one comment and conflicts also need resolved

File: .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml (hunk 1 above)
Comment (andreydevyatkin) [reply .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml:69]: @damccorm Does it make sense? We currently use strict checking to tell whether a comment is as expected or not (e.g. https://github.com/apache/beam/pull/28352/files#diff-22ba8e2fcd77c357a00841ff632eb4c174f1b3aee4863782ff150e9b27e609aaR65). If we want to handle extra whitespace/other chars I think we need to update all workflows in a separate PR.

Comment (aleksandr-dudko) [reply .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml]: It's done.

Comment (damccorm) [reply .github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml]: Ah you're right that we do this with other workflows, though we do use startswith for other matrix jobs...

With that in mind, I'm good with either approach here (endswith or contains), so we can just leave it as is for this one.

**Review by damccorm — APPROVED** [review]

**Pull request merged**